Documentation/devel/architecture.md

```diff
@@ -102,3 +101,63 @@ This will activate all the reaper services when one of the targets is activated,
 ### Stage 2
 
 The final stage, stage2, is the actual environment in which the applications run, as launched by stage1.
+
+## Process execution chain
+
+After calling `rkt` the execution chain follows the numbering of stages, having the following general order:
+
+![execution-flow](execution-flow.png)
+
+1. invoking process -> stage0:
+The invoking process uses its own mechanism to invoke the rkt binary (stage0). When started via a regular shell or a supervisor, stage0 is usually forked and exec'ed becoming a child process of the invoking shell or supervisor.
+
+2. stage0 -> stage1:
+An ordinary `exec(3)` is being used to replace the stage0 process with the stage1 entrypoint. The entrypoint is referenced by the `coreos.com/rkt/stage1/run` annotation in the stage1 image manifest.
+
+3. stage1 -> stage2:
+The stage1 entrypoint uses its mechanism to invoke the stage2 app executables. The app executables are referenced by the `apps.app.exec` settings in the stage2 image manifest.
+
+The details of the execution flow varies across the different stage1 implementations.
+
+### systemd/nspawn based flavors
+
+The "host", "src", and "coreos" flavors (referenced to as systemd/nspawn flavors) use  `systemd-nspawn`, and `systemd` to set up the execution chain.
+
+As described above the rkt binary (stage0) is usually forked and exec'ed in the invoking process. The entrypoint for stage1 is implemented in the `stage1/init/init.go` binary and sets up the following execution chain:
+
+1. "ld-linux-*.so.*": Depending on the architecture the appropriate loader helper in the stage1 rootfs is invoked using "exec". This makes sure that subsequent binaries load shared libraries from the stage1 rootfs and not from the host file system.
+
+2. "systemd-nspawn": Used for starting the actual container. systemd-nspawn registers the started container in "systemd-machined", if available. It is parametrized with the `--boot` option to instruct it to "fork+exec" systemd as the supervisor in the started container.
+
+3. "systemd": Used as the supervisor in the started container. Similar as on a regular host system, it uses "fork+exec" to execute the child app processes.
+
```

Comment:
Not sure on the level of details objective but, more than fork+exec I'll detail that it setups various systemd units per pod and per app (prepare, the app service, reaper), that the app is chrooted using the systemd unit RootDirectory and many other things.